hlsp/hlsp.go

```diff
@@ -1,46 +1,45 @@
 package hlsp
 
 import (
-	"fmt"
+	"encoding/json"
 
+	"github.com/ghodss/yaml"
 	"github.com/xeipuuv/gojsonschema"
 )
 
-// Parse receives either a YAML or JSON AsyncAPI document.
+// Parse receives either a YAML or JSON AsyncAPI document, and tries to parse it.
+func Parse(yamlOrJSONDocument []byte) (json.RawMessage, *ParserError) {
+	jsonDocument, err := yaml.YAMLToJSON(yamlOrJSONDocument)
```

Comment:
the call to YAMLToJSON looks like its always doing a conversion to JSON from yaml but it is not. it only does it if the content is json.

Reply:
What do you mean? It's being tested here and it works: https://github.com/asyncapi/parser/pull/18/files#diff-fee0983ac72dab9799141e78d9c594f9R47

Reply:
sorry, I meant the name, but I guess I'm been too picky  :)

Reply:
I'm not understanding 😅

Reply:
haha, no worries its fine. this looks good to me